### PR TITLE
Fix `SliverAppBar.medium` & `SliverAppBar.large` text scale

### LINKED ISSUE #114340: SliverAppBar.medium cuts off title when the system font size is increased
When the system font size has been increased, the title is cut off in a `SliverAppBar.medium`, similar to this issue: https://github.com/flutter/flutter/issues/104493

# repro

- Run the sliver app bar medium example
- Increase system font size

https://api.flutter.dev/flutter/material/SliverAppBar/SliverAppBar.medium.html

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/114340

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

/// Flutter code sample for [SliverAppBar.medium].

void main() {
  runApp(const AppBarMediumApp());
}

class AppBarMediumApp extends StatelessWidget {
  const AppBarMediumApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        useMaterial3: true,
      ),
      home: MediaQuery(
        data: MediaQuery.of(context).copyWith(textScaleFactor: 3.0),
        child: Material(
          child: CustomScrollView(
            slivers: <Widget>[
              SliverAppBar.medium(
                leading:
                    IconButton(icon: const Icon(Icons.menu), onPressed: () {}),
                title: const Text('Medium App Bar'),
                actions: <Widget>[
                  IconButton(icon: const Icon(Icons.more_vert), onPressed: () {}),
                ],
              ),
              // SliverAppBar.large(
              //   leading:
              //       IconButton(icon: const Icon(Icons.menu), onPressed: () {}),
              //   title: const Text('Large App Bar'),
              //   actions: <Widget>[
              //     IconButton(
              //         icon: const Icon(Icons.more_vert), onPressed: () {}),
              //   ],
              // ),
              // Just some content big enough to have something to scroll.
              SliverToBoxAdapter(
                child: Card(
                  child: SizedBox(
                    height: 1200,
                    child: Padding(
                      padding: const EdgeInsets.fromLTRB(8, 100, 8, 100),
                      child: Text(
                        'Here be scrolling content...',
                        style: Theme.of(context).textTheme.headlineSmall,
                      ),
                    ),
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

``` 
	
</details>

### Before
|  Medium App Bar - `textScaleFactor: 3.0` |   Large App Bar - `textScaleFactor.30` |
| --------------- | --------------- |
|  |  |

### After
|  Medium App Bar - `textScaleFactor: 3.0` |   Large App Bar - `textScaleFactor.30` |
| --------------- | --------------- |
|  |  |

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
Interesting! I did not know _kMaxTitleTextScaleFactor existed.
Is this in the spec? I see there is a TODO comment about documenting this that refers to https://github.com/flutter/flutter/issues/58769

Can you add some mention of this text scale limit to the AppBar docs?

**Comment by TahaTesser** [comment]
> Is this in the spec? I see there is a TODO comment about documenting this that refers to #58769

It's not but the scale factor is relatively the same regular app bar when compared with native 
(which also updates the position of the title so it doesn't cut off, which took me quite a while to figure out by testing native a lot)

Interesting! I did not know _kMaxTitleTextScaleFactor existed.
Is this in the spec? I see there is a TODO comment about documenting this that refers to https://github.com/flutter/flutter/issues/58769

> Can you add some mention of this text scale limit to the AppBar docs?

Sure.  :)

**Comment by TahaTesser** [comment]
@Piinks 
I need to make a similar PR for `NavigationRail`. 
I noticed the same lack of information there. The accessibility pages have no information about the text scale e.g https://m3.material.io/components/top-app-bar/accessibility. 
I've to rely on existing scaling values in Flutter or test the native side myself to compare.

File: packages/flutter/test/material/app_bar_test.dart [hunk 1]
```diff
@@ -4649,6 +4650,190 @@ void main() {
...
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildAppBar());
+
+    // TODO(tahatesser): https://github.com/flutter/flutter/issues/99933
+    //                A bug in the HTML renderer and/or Chrome 96+ causes a
+    //                discrepancy in the paragraph height.
```
Comment (Piinks) [inline packages/flutter/test/material/app_bar_test.dart:4721]: There is a strange indentation here

File: packages/flutter/test/material/app_bar_test.dart [hunk 2]
```diff
@@ -4649,6 +4650,190 @@ void main() {
...
+              ],
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildAppBar());
+    // TODO(tahatesser): https://github.com/flutter/flutter/issues/99933
+    //                A bug in the HTML renderer and/or Chrome 96+ causes a
```
Comment (Piinks) [inline packages/flutter/test/material/app_bar_test.dart:4813]: Same here

File: packages/flutter/test/material/app_bar_test.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/test/material/app_bar_test.dart:4721]: Oh I see, It looks like the comment is aligned withe the end of the todo colon. Not necessary, but I don't see why not. :)

**Review by Piinks — APPROVED** [review]
LGTM

File: packages/flutter/test/material/app_bar_test.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/test/material/app_bar_test.dart]: I copied this comment from another class. It looks strange for sure. Fixed.

**Pull request merged**